sechub-web-ui-solution/helm/sechub-web-ui/templates/deployment.yaml

```
@@ -0,0 +1,90 @@
+# SPDX-License-Identifier: MIT
+
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ .Values.name }}
+spec:
+  replicas: {{ .Values.replicaCount }}
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxSurge: 1
+      maxUnavailable: 49%
+  selector:
+    matchLabels:
+      name: {{ .Values.name }}
+  template:
+    metadata:
+      labels:
+        name: {{ .Values.name }}
+{{- if .Values.templateMetadataAnnotations }}
+      annotations:
+        {{ .Values.templateMetadataAnnotations | indent 8 | trim }}
+{{- end }}
+    spec:
+      securityContext:
+        runAsUser: 4242  # uid of the application user. (should be same as in the Dockerfile)
+        runAsGroup: 4242 # gid of the application group. (should be same as in the Dockerfile)
+        fsGroup: 4242    # This gid has write access to the mounted volumes.
+{{- if .Values.go_mmproxy.enabled }}
+      initContainers:
+        - name: setup
+          image: "{{ .Values.go_mmproxy.image }}"
+          command:
+            - "/bin/sh"
+            - "-cx"
+          args:
+            - |
+              /sbin/ip rule add from 127.0.0.1/8 iif lo table 123
+              /sbin/ip route add local 0.0.0.0/0 dev lo table 123
+          securityContext:
+            capabilities:
+              add:
+                - NET_ADMIN
+{{- end }}
+      containers:
+{{- if .Values.go_mmproxy.enabled }}
+      # go-mmproxy sidecar container
+      - name: go-mmproxy
+        image: "{{ .Values.go_mmproxy.image }}"
+        ports:
+          - containerPort: 8080
+        securityContext:
+          capabilities:
+            add:
+              - NET_ADMIN
+        args:
+          - "-v=0"           # loglevel 0 - no logging ; 1 - log errors ; 2 - log all state changes
+          - "-l"             # listen
+          - "0.0.0.0:8080"   # on port 8080
+          - "-4"             # tcp v4
+          - "127.0.0.1:4443" # forward to SecHub WebUI container (localhost port 4443)
+{{- end }}
+
+        # Web-UI container
+        - name: {{ .Values.name }}
+          image: {{ .Values.image }}
+          labels:
+            name: {{ .Values.name }}
+          ports:
+          - name: {{ .Values.name }}
+            containerPort: 8443
```

We should not forget to define startup/liveness/readiness probes.
This is just an example with `/alive-test.html` which may be added to the container image as a tiny file.
```suggestion
          containerPort: 8443
        startupProbe:
          httpGet:
            scheme: HTTPS
            path: /alive-test.html
            port: {{ .Values.name }}-https
          failureThreshold: 24
          periodSeconds: 5
          successThreshold: 1
          timeoutSeconds: 1
          initialDelaySeconds: 5
        livenessProbe:
          httpGet:
            scheme: HTTPS
            path: /alive-test.html
            port: {{ .Values.name }}-https
          failureThreshold: 3
          periodSeconds: 10
          successThreshold: 1
          timeoutSeconds: 3
```